### PR TITLE
Sentry return

### PR DESCRIPTION
**Pull request by nasenbaer13**
### What does this PR do?

- add transport option (sentry configuration)
- aggregation for job returns, remove jid and create stable messages instead
- consider result as success if no state fails
- minor fixes

### What issues does this PR fix or reference?

see commit messages and above.

### Tests written?

No

### CODE REVIEW

**Comment by nasenbaer13** [comment]
Please merge back to 2017.7 if possible.

**Comment by cachedout** [comment]
@nasenbaer13 Would you be willing to add a unit test for the sentry returner so that we can cover the `get_message` function?

**Comment by cachedout** [comment]
@nasenbaer13 We're waiting to merge this until we can get a test case along with it. Will you have time to come back to this? Please let us know. Thanks.

**Comment by nasenbaer13** [comment]
yes, we'll add a test as soon as we can.

**Comment by rrroo** [comment]
@cachedout I added a test for get_message. Please let me know, if there are more changes needed.

**Comment by rrroo** [comment]
@cachedout  @rallytime Please let me know if there is more to do to get this merged.

File: tests/unit/returners/test_sentry.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Test Sentry Returner
+    '''
+    ret = {'id': '12345',
+           'fun': 'mytest.func',
+           'fun_args': ['arg1', 'arg2', {'foo': 'bar'}],
+           'jid': '54321',
+           'return': 'Long Return containing a Traceback'}
+
+    def test_get_message(self):
+        assert sentry._get_message(self.ret) == 'salt func: mytest.func arg1 arg2 foo=bar'
```
Comment (rallytime) [inline tests/unit/returners/test_sentry.py]: Hey @nasenbaer13 - Unfortunately we haven't made the full switch over to pytest. Can you change these assertions from `assert` to the [unittest assertion types](https://docs.python.org/2/library/unittest.html#assert-methods) such as `assertEqual`?

Comment (rrroo) [reply tests/unit/returners/test_sentry.py]: Done

**Comment by rrroo** [comment]
@rallytime I believe the test failure is unrelated. Please let me know if there are other things to change.

**Comment by rrroo** [comment]
@cachedout @rallytime Any update on this PR?

**Comment by rallytime** [comment]
@rrada Sorry for the delay on this. We just need to get this reviewed, and then we can get it in.

@garethgreenaway Maybe this is one you could look over.

**Review by garethgreenaway — APPROVED** [review]

**Comment by rrroo** [comment]
Thanks @garethgreenaway and @rallytime for the review and merge!

**Pull request merged**